packages/cardpay-subgraph/src/mappings/reward-pool.ts

```diff
@@ -72,3 +76,23 @@ export function handleRewardTokensAdded(event: RewardTokensAdded): void {
   entity.blockNumber = event.block.number;
   entity.save();
 }
+
+export function handleMerkleRootSubmission(event: MerkleRootSubmission): void {
+  let txnHash = event.transaction.hash.toHex();
+  let rewardProgramID = toChecksumAddress(event.params.rewardProgramID);
+  let paymentCycle = event.params.paymentCycle;
+
+  let rewardProgramEntity = RewardProgram.load(rewardProgramID);
+  if (rewardProgramEntity == null) {
+    log.warning(
+      'Cannot process tokens added txn {}: Reward program entity does not exist for safe address {}. This is likely due to the subgraph having a startBlock that is higher than the block the safe was created in.',
+      [txnHash, rewardProgramID]
+    );
+    return;
+  }
+
+  let entity = new MerkleRoot(txnHash);
+  entity.rewardProgram = rewardProgramID;
```

Comment:
Should this be 

    entity.rewardProgram = rewardProgramEntity

?

Reply:
No I don't think so. You can see multiple examples 
- [here](https://github.com/cardstack/cardstack/blob/73e6b58d4b862229097c941be53b9edfdfda36ab/packages/cardpay-subgraph/src/mappings/prepaid-card.ts#L63)
- [here](https://github.com/cardstack/cardstack/blob/73e6b58d4b862229097c941be53b9edfdfda36ab/packages/cardpay-subgraph/src/mappings/merchant-manager.ts#L26)

This is a pattern I've followed. But there was never convincing documentation to point me to the correctness of this pattern.